pennylane/gradients/parameter_shift.py

```diff
@@ -0,0 +1,514 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for computing the parameter-shift gradient
+of a qubit-based quantum tape.
+"""
+# pylint: disable=protected-access,too-many-arguments
+import numpy as np
+
+import pennylane as qml
+
+from .finite_difference import finite_diff, generate_shifted_tapes
+
+
+NONINVOLUTORY_OBS = {
+    "Hermitian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),
+    "SparseHamiltonian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),

```

Comment:
Oh, just realized it seems like variance does not apply for [SparseHamiltonian at the moment](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/ops/qubit.py#L3255):
```
    .. warning::
        ``SparseHamiltonian`` observables can only be used to return expectation values.
        Variances and samples are not supported.
```

Maybe worth still leaving this as is once the support is in? :thinking:

Reply:
Yep, I left it here just for the future :)